### PR TITLE
feat(ecs-pattens): pass healthy percent & deregistration delay params

### PR DESCRIPTION
**Pull request by cheruvian**
The `ecs-patterns` for multiple target groups expose a subset of the available properties for each of the lower level constructs form which it is composed. This subset seems to be manually curated today, however it excludes a number of production critical properties.

This change adds 2 such important properties that I have needed recently.
* `minHealthyPercent` from `AWS::ECS::Service`
* `deregistrationDelay` from `AWS::ElasticLoadBalancingV2::TargetGroup`'s `TargetGroupAttributes`

I believe it may be possible/beneficial to operate on a block list rather than allow list of properties and include all the properties of the included child constructs (including docs, etc...) except for those that don't make sense in a multiple target group context. However, it wasn't clear if this change would be approved and as such I have considered this out of scope for this change, if that is acceptable or desired, I propose that as a different / additional scope of work.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ecs-patterns/README.md
Comment (kylelaker) [inline packages/aws-cdk-lib/aws-ecs-patterns/README.md:90]:
```suggestion
      deregistrationDelay: Duration.seconds(300),
```

File: packages/aws-cdk-lib/aws-ecs-patterns/README.md
Comment (kylelaker) [inline packages/aws-cdk-lib/aws-ecs-patterns/README.md:117]:
```suggestion
      deregistrationDelay: Duration.seconds(300),
```

**Review by kylelaker — COMMENTED** [review]
Providing a meaningful review is challenging while the build is failing. Reviewing the build logs, applying these changes to the `README` should fix the failing build.

File: packages/aws-cdk-lib/aws-ecs-patterns/README.md
Comment (cheruvian) [reply packages/aws-cdk-lib/aws-ecs-patterns/README.md]: Apologies, I couldn't get the tests running on my machine even before changes. Various things complained and/or timed out.

Updated with your corrections.

Comment (cheruvian) [reply packages/aws-cdk-lib/aws-ecs-patterns/README.md]: FWIW confirmed the change itself works by using `patch-package`, though obviously understand we need the tests to pass.

**Comment by cheruvian** [comment]
Any thoughts on this now that the build is passing @kylelaker?

**Comment by cheruvian** [comment]
Found another one that I need today

`stickinessCookieDuration`

Can someone from AWS please advise if `ApplicationTargetProps` should just extend or otherwise delegate `AddApplicationTargetsProps` since that seems to be a majority of it's function. This would also allow for udpates to ALB Target Groups to be taken for 'free' ™ by the ecs patterns

File: packages/aws-cdk-lib/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts
Comment (msambol) [inline packages/aws-cdk-lib/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts:116]: could you add an `@see` here with a link to the docs?

Comment (msambol) [reply packages/aws-cdk-lib/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts:116]: also, I think I'd prefer this to be `minimumHealthyPercent` to match CloudFormation.

Comment (msambol) [reply packages/aws-cdk-lib/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts]: @cheruvian I didn't see it in `BaseServiceOptions`. Apologies, you can change it to make that.

Comment (msambol) [reply packages/aws-cdk-lib/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts]: Rest LGTM, if you want to change this back.

Comment (cheruvian) [reply packages/aws-cdk-lib/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts]: Any thoughts on whether this should actually just `extend Omit<BaseServiceOptions, ...>`?

Comment (msambol) [reply packages/aws-cdk-lib/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts:116]: I think this is fine but we'll see what the maintainer's say.

File: packages/aws-cdk-lib/aws-ecs-patterns/lib/base/application-multiple-target-groups-service-base.ts
Comment (msambol) [inline packages/aws-cdk-lib/aws-ecs-patterns/lib/base/application-multiple-target-groups-service-base.ts:297]: This should actually be `Stickiness is disabled`. See https://github.com/aws/aws-cdk/pull/29726

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/ec2/integ.multiple-application-load-balanced-ecs-service.ts
Comment (msambol) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/ec2/integ.multiple-application-load-balanced-ecs-service.ts:27]: can you add an integration test for NLB as well?

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
A couple things here:

Firstly +1 to the comments inline. 

Secondly, from the amounts of updates in the integ test files, it looks like they were manually updated and that the test wasn't actually run. Please remove the manual updates and run the test with the `--update-on-failed` flag and then add all the resulting changes to this PR.

**Pull request closed without merging**
